backend/app/controllers/projects.js

```diff
@@ -321,25 +322,93 @@ module.exports = {
             req.body = _.extend(req.body, {
                 userAdminId: req.user.realmUserId
             }); // add from realmUserId instead of user id
-            var result = yield thunkQuery(
+            var result = _.first(yield thunkQuery(
                 Project
                 .insert(_.pick(req.body, Project.table._initialConfig.columns))
                 .returning(Project.id)
-            );
+            ));
+
+            result.name = req.body.codeName;
+            result.status = 0;
+
+            // Having it automatically insert into products and workflows for now.
+            result.productId = _.first(yield thunkQuery(
+                Product.insert({
+                    title: result.name,
+                    description: req.body.description,
+                    projectId: result.id,
+                    status: 0,
+                }).returning(Product.id)
+            )).id;
+
+            result.workflowId = _.first(yield thunkQuery(
+                Workflow.insert({
+                    name: result.name,
+                    description: req.body.description,
+                    productId: result.productId,
+                }).returning(Workflow.id)
+            )).id;
+
+            result.users = [];
+            result.stages = [];
+            result.userGroups = [];
+            result.subjects = [];
+
             return result;
         }).then(function (data) {
             bologger.log({
                 req: req,
                 user: req.user,
                 action: 'insert',
                 object: 'projects',
-                entity: _.first(data).id,
+                entity: data.id,
                 info: 'Add new project'
             });
-            res.status(201).json(_.first(data));
+            res.status(201).json(data);
+        }, function (err) {
+            next(err);
+        });
+    },
+
+    userAssignment: function (req, res, next) {
+        co(function* () {
+            yield * common.insertProjectUser(req, req.body.userId, req.body.projectId);
+            return true;
```

Comment:
Instead of returning `true` here, we should return the data that was inserted and some helpful message

Reply:
Create a PR for it.